retroshare-gui/src/retroshare-gui.pro

```diff
@@ -1391,42 +1408,43 @@ gxsgui {
-    ../../supportlibs/cmark/src/render.h								 \
-    ../../supportlibs/cmark/src/scanners.h							 \
-    ../../supportlibs/cmark/src/utf8.h									 \
-
-  SOURCES += \
-    ../../supportlibs/cmark/src/blocks.c									 \
-    ../../supportlibs/cmark/src/buffer.c									 \
-    ../../supportlibs/cmark/src/cmark.c										 \
-    ../../supportlibs/cmark/src/cmark_ctype.c							 \
-    ../../supportlibs/cmark/src/commonmark.c							 \
-    ../../supportlibs/cmark/src/houdini_href_e.c					 \
-    ../../supportlibs/cmark/src/houdini_html_e.c					 \
-    ../../supportlibs/cmark/src/houdini_html_u.c					 \
-    ../../supportlibs/cmark/src/html.c										 \
-    ../../supportlibs/cmark/src/inlines.c									 \
-    ../../supportlibs/cmark/src/iterator.c								 \
-    ../../supportlibs/cmark/src/latex.c										 \
-    ../../supportlibs/cmark/src/man.c											 \
-    ../../supportlibs/cmark/src/node.c										 \
-    ../../supportlibs/cmark/src/references.c							 \
-    ../../supportlibs/cmark/src/render.c									 \
-    ../../supportlibs/cmark/src/scanners.c								 \
-    ../../supportlibs/cmark/src/utf8.c										 \
-    ../../supportlibs/cmark/src/xml.c											 \
-
+	DEFINES *= USE_CMARK
+
+	cmark_builtin {
+		HEADERS += \
+		  ../../supportlibs/cmark/src/buffer.h								 \
+		  ../../supportlibs/cmark/src/chunk.h									 \
+		  ../../supportlibs/cmark/src/cmark.h									 \
+		  ../../supportlibs/cmark/src/cmark_ctype.h						 \
+		  ../../supportlibs/cmark/src/houdini.h								 \
+		  ../../supportlibs/cmark/src/inlines.h								 \
+		  ../../supportlibs/cmark/src/iterator.h							 \
+		  ../../supportlibs/cmark/src/node.h									 \
+		  ../../supportlibs/cmark/src/parser.h								 \
+		  ../../supportlibs/cmark/src/references.h						 \
+		  ../../supportlibs/cmark/src/render.h								 \
+		  ../../supportlibs/cmark/src/scanners.h							 \
+		  ../../supportlibs/cmark/src/utf8.h									 \
+
+		SOURCES += \
+		  ../../supportlibs/cmark/src/blocks.c									 \
+		  ../../supportlibs/cmark/src/buffer.c									 \
+		  ../../supportlibs/cmark/src/cmark.c										 \
+		  ../../supportlibs/cmark/src/cmark_ctype.c							 \
+		  ../../supportlibs/cmark/src/commonmark.c							 \
+		  ../../supportlibs/cmark/src/houdini_href_e.c					 \
+		  ../../supportlibs/cmark/src/houdini_html_e.c					 \
+		  ../../supportlibs/cmark/src/houdini_html_u.c					 \
+		  ../../supportlibs/cmark/src/html.c										 \
+		  ../../supportlibs/cmark/src/inlines.c									 \
+		  ../../supportlibs/cmark/src/iterator.c								 \
+		  ../../supportlibs/cmark/src/latex.c										 \
+		  ../../supportlibs/cmark/src/man.c											 \
+		  ../../supportlibs/cmark/src/node.c										 \
+		  ../../supportlibs/cmark/src/references.c							 \
+		  ../../supportlibs/cmark/src/render.c									 \
+		  ../../supportlibs/cmark/src/scanners.c								 \
+		  ../../supportlibs/cmark/src/utf8.c										 \
+		  ../../supportlibs/cmark/src/xml.c											 \
+	}
```

Comment:
I think this should be avoided, seems cmark uses cmake as build system, so it can be built with it's own build system as we do for restbed

Reply:
Like you've done here?
https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/libretroshare.pro#L839
But this is done on every case, even if library is installed by system?

Reply:
yeah like that, but in your case do it only when cmark is not installed in the system